<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">

<title>Single cell RNA-seq (ES mouse) using Seurat</title>
<meta name="description" content="Data analysis and visualisations">

<link rel="stylesheet" href="/css/main.css">
<link rel="canonical" href="http://jespermaag.github.io/2018-03-25-SingleCellESmouse%20copy/">
<link rel="alternate" type="application/rss+xml" title="Jesper Maag PhD" href="http://jespermaag.github.io/feed.xml" />

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', '', 'auto');
  ga('send', 'pageview');
</script>


</head>
<body>
  <header class="site-header">
  <div class="container">
    <input type="checkbox" id="toggleNavbar">
    <nav class="navbar">
      <ul>
        <li><a href="/" title="Home">Home</a></li>
        
          <li><a href="/about" title="About">About</a></li>
        
          <li><a href="/education" title="Education">Education</a></li>
        
          <li><a href="/experience" title="Experience">Experience</a></li>
        
          <li><a href="/blog" title="Blog">Blog</a></li>
        
          <li><a href="/publications" title="Publications">Publications</a></li>
        
      </ul>
    </nav>
  </div>
</header>


<main class="main-container">
  <div class="container">
    <article role="article" class="post">

  <div class="card">
    <header class="post-header">
      <h1 class="post-title">Single cell RNA-seq (ES mouse) using Seurat</h1>
      <em class="post-meta">
        <time>Mar 25, 2018</time>
      </em>
    </header>

    <div class="post-content">
      
        <figure class="post-thumbnail ">
          <img src="/assets/Rfig/subsetPCA-2.svg" alt="Single cell RNA-seq (ES mouse) using Seurat">
        </figure>
      
      <h2 id="in-this-post-i-will-try-to-analyze-single-cell-rna-seq-data-using-seurat">In this post I will try to analyze single cell RNA-seq data using Seurat</h2>
<p>Data was downlaoded from https://github.com/debsin/dropClust</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span><span class="kn">library</span><span class="p">(</span>RCurl<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>Seurat<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>cellrangerRkit<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>data.table<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>tidyverse<span class="p">)</span></code></pre></figure>

<p>This post also follows examples from https://davetang.org/muse/2017/08/01/getting-started-seurat/
First we remove all genes that are expressed in less than 3 cells.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>esMus <span class="o">&lt;-</span> Read10X<span class="p">(</span><span class="kt">c</span><span class="p">(</span><span class="s">&#39;~/github/jespermaag.github.io/es_mouse/&#39;</span><span class="p">))</span>

<span class="kp">dim</span><span class="p">(</span>esMus<span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text"><span></span>## [1] 24175  2717</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>keep <span class="o">&lt;-</span> <span class="kp">apply</span><span class="p">(</span>esMus<span class="p">,</span> <span class="m">1</span><span class="p">,</span> <span class="kr">function</span><span class="p">(</span>x<span class="p">)</span> <span class="kp">sum</span><span class="p">(</span>x<span class="o">&gt;</span><span class="m">0</span><span class="p">))</span>
<span class="kp">table</span><span class="p">(</span>keep<span class="o">&gt;=</span><span class="m">3</span><span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text"><span></span>## 
## FALSE  TRUE 
##   153 24022</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>keep <span class="o">&lt;-</span> keep <span class="o">&gt;=</span> <span class="m">3</span>

esMus <span class="o">&lt;-</span> esMus<span class="p">[</span>keep<span class="p">,]</span>

<span class="kp">summary</span><span class="p">(</span><span class="kp">colSums</span><span class="p">(</span>esMus<span class="p">))</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text"><span></span>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##    1647    6263   17373   20033   31086   85686</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>genes <span class="o">&lt;-</span> <span class="kp">apply</span><span class="p">(</span>esMus<span class="p">,</span> <span class="m">2</span><span class="p">,</span> <span class="kr">function</span><span class="p">(</span>x<span class="p">)</span> <span class="kp">sum</span><span class="p">(</span>x<span class="o">&gt;</span><span class="m">0</span><span class="p">))</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>genes <span class="o">%&gt;%</span> 
    melt<span class="p">()</span> <span class="o">%&gt;%</span>
    ggplot<span class="p">(</span>aes<span class="p">(</span>x<span class="o">=</span>value<span class="p">))</span> <span class="o">+</span>
        geom_histogram<span class="p">(</span>colour<span class="o">=</span><span class="s">&#39;black&#39;</span><span class="p">,</span> fill<span class="o">=</span><span class="s">&quot;steelblue&quot;</span><span class="p">)</span> <span class="o">+</span>
        theme_classic<span class="p">()</span><span class="o">+</span>
        xlab<span class="p">(</span><span class="s">&#39;Number genes (1&gt;= counts)&#39;</span><span class="p">)</span> <span class="o">+</span>
        ggtitle<span class="p">(</span><span class="s">&#39;Number of genes detected per cell&#39;</span><span class="p">)</span><span class="o">+</span>
	    theme<span class="p">(</span> axis.title.x <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">14</span><span class="p">),</span> axis.text.x <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">12</span><span class="p">,</span>colour<span class="o">=</span><span class="s">&#39;black&#39;</span><span class="p">),</span>  
        axis.text.y <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">12</span><span class="p">,</span>colour<span class="o">=</span><span class="s">&#39;black&#39;</span><span class="p">),</span> axis.title.y <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">14</span><span class="p">))</span></code></pre></figure>

<p><img src="/assets/Rfig/explorePlots-1.svg" alt="plot of chunk explorePlots" /></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span><span class="kp">colSums</span><span class="p">(</span>esMus<span class="p">)</span> <span class="o">%&gt;%</span>
    melt<span class="p">()</span> <span class="o">%&gt;%</span>
    ggplot<span class="p">(</span>aes<span class="p">(</span>x<span class="o">=</span>value<span class="p">))</span> <span class="o">+</span>
        geom_histogram<span class="p">(</span>colour<span class="o">=</span><span class="s">&#39;black&#39;</span><span class="p">,</span> fill<span class="o">=</span><span class="s">&quot;steelblue&quot;</span><span class="p">)</span> <span class="o">+</span>
        theme_classic<span class="p">()</span><span class="o">+</span>
        xlab<span class="p">(</span><span class="s">&#39;Counts per cell)&#39;</span><span class="p">)</span> <span class="o">+</span>
        ggtitle<span class="p">(</span><span class="s">&#39;Total counts per cell&#39;</span><span class="p">)</span><span class="o">+</span>
        theme<span class="p">(</span> axis.title.x <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">14</span><span class="p">),</span> axis.text.x <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">12</span><span class="p">,</span>colour<span class="o">=</span><span class="s">&#39;black&#39;</span><span class="p">),</span>  
        axis.text.y <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">12</span><span class="p">,</span>colour<span class="o">=</span><span class="s">&#39;black&#39;</span><span class="p">),</span> axis.title.y <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">14</span><span class="p">))</span></code></pre></figure>

<p><img src="/assets/Rfig/explorePlots-2.svg" alt="plot of chunk explorePlots" /></p>

<p>As seen in the number of gene plots, there is a binomial distribution of number of genes per cell.
I will keep this in mind for further analysis. 
The binomial distribtion is missing from the total counts per cell.</p>

<p>To use Seurat, I first have to create a Seurat object</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>esMusSeur <span class="o">&lt;-</span> CreateSeuratObject<span class="p">(</span>raw.data <span class="o">=</span> esMus<span class="p">,</span>
                           min.cells <span class="o">=</span> <span class="m">3</span><span class="p">,</span>
                           min.genes <span class="o">=</span> <span class="m">100</span><span class="p">,</span>
                           project <span class="o">=</span> <span class="s">&quot;ES_mouse&quot;</span><span class="p">)</span>

esMusSeur</code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text"><span></span>## An object of class seurat in project ES_mouse 
##  24022 genes across 2717 samples.</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span><span class="kp">dim</span><span class="p">(</span>esMusMat<span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text"><span></span>## [1] 24047  2717</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>slotNames<span class="p">(</span>esMusSeur<span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text"><span></span>##  [1] &quot;raw.data&quot;     &quot;data&quot;         &quot;scale.data&quot;   &quot;var.genes&quot;   
##  [5] &quot;is.expr&quot;      &quot;ident&quot;        &quot;meta.data&quot;    &quot;project.name&quot;
##  [9] &quot;dr&quot;           &quot;assay&quot;        &quot;hvg.info&quot;     &quot;imputed&quot;     
## [13] &quot;cell.names&quot;   &quot;cluster.tree&quot; &quot;snn&quot;          &quot;calc.params&quot; 
## [17] &quot;kmeans&quot;       &quot;spatial&quot;      &quot;misc&quot;         &quot;version&quot;</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>esMusSeur <span class="o">&lt;-</span> NormalizeData<span class="p">(</span>object <span class="o">=</span> esMusSeur<span class="p">,</span>
                      normalization.method <span class="o">=</span> <span class="s">&quot;LogNormalize&quot;</span><span class="p">,</span>
                      scale.factor <span class="o">=</span> <span class="m">1e4</span><span class="p">)</span>



 

<span class="kp">colSums</span><span class="p">(</span>esMusSeur<span class="o">@</span>data<span class="p">)</span> <span class="o">%&gt;%</span>
    melt<span class="p">()</span> <span class="o">%&gt;%</span>
    ggplot<span class="p">(</span>aes<span class="p">(</span>x<span class="o">=</span>value<span class="p">))</span> <span class="o">+</span>
        geom_histogram<span class="p">(</span>colour<span class="o">=</span><span class="s">&#39;black&#39;</span><span class="p">,</span> fill<span class="o">=</span><span class="s">&quot;steelblue&quot;</span><span class="p">)</span> <span class="o">+</span>
        theme_classic<span class="p">()</span><span class="o">+</span>
        xlab<span class="p">(</span><span class="s">&#39;Normalised counts per cell&#39;</span><span class="p">)</span> <span class="o">+</span>
        ggtitle<span class="p">(</span><span class="s">&#39;Total counts per cell&#39;</span><span class="p">)</span><span class="o">+</span>
        theme<span class="p">(</span> axis.title.x <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">14</span><span class="p">),</span> axis.text.x <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">12</span><span class="p">,</span>colour<span class="o">=</span><span class="s">&#39;black&#39;</span><span class="p">),</span>  
        axis.text.y <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">12</span><span class="p">,</span>colour<span class="o">=</span><span class="s">&#39;black&#39;</span><span class="p">),</span> axis.title.y <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">14</span><span class="p">))</span></code></pre></figure>

<p><img src="/assets/Rfig/createSeurat-1.svg" alt="plot of chunk createSeurat" />
As seen in the previous figure, after normalised the binomial distribution appears for the total counts.</p>

<p>Below I want to find the variable genes between the cells and plot the PCA.
Since this is my first single cell analysis, I will use the parameters from https://davetang.org/muse/2017/08/01/getting-started-seurat/</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>esMusSeur <span class="o">&lt;-</span> FindVariableGenes<span class="p">(</span>object <span class="o">=</span> esMusSeur<span class="p">,</span>
                          mean.function <span class="o">=</span> ExpMean<span class="p">,</span>
                          dispersion.function <span class="o">=</span> LogVMR<span class="p">,</span>
                          x.low.cutoff <span class="o">=</span> <span class="m">0.0125</span><span class="p">,</span>
                          x.high.cutoff <span class="o">=</span> <span class="m">3</span><span class="p">,</span>
                          y.cutoff <span class="o">=</span> <span class="m">0.5</span><span class="p">)</span></code></pre></figure>

<p><img src="/assets/Rfig/PCA-1.svg" alt="plot of chunk PCA" /></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span><span class="kp">length</span><span class="p">(</span>esMusSeur<span class="o">@</span>var.genes<span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text"><span></span>## [1] 5707</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span><span class="kp">head</span><span class="p">(</span>esMusSeur<span class="o">@</span>hvg.info<span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text"><span></span>##        gene.mean gene.dispersion gene.dispersion.scaled
## Gm7102 0.5832131        5.417292              16.151102
## Dcdc2c 0.4870505        5.006380              18.885776
## Rn28s1 2.9695050        4.793843               2.929660
## Rn4.5s 2.7443443        4.607135               3.618150
## Rn45s  3.6240987        4.024695               2.001075
## Ccdc36 3.3616208        3.766651               1.659102</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>esMusSeur<span class="o">@</span>hvg.info <span class="o">%&gt;%</span>
    ggplot<span class="p">(</span>aes<span class="p">(</span> x<span class="o">=</span> gene.mean<span class="p">,</span> y <span class="o">=</span> gene.dispersion.scaled<span class="p">))</span> <span class="o">+</span>
        geom_point<span class="p">(</span>pch<span class="o">=</span><span class="m">21</span><span class="p">,</span> fill<span class="o">=</span><span class="s">&#39;steelblue&#39;</span><span class="p">)</span> <span class="o">+</span>
        theme_classic<span class="p">()</span><span class="o">+</span>
        xlab<span class="p">(</span><span class="s">&#39;Mean gene expression&#39;</span><span class="p">)</span> <span class="o">+</span>
        ylab<span class="p">(</span><span class="s">&#39;scaled dispersion&#39;</span><span class="p">)</span></code></pre></figure>

<p><img src="/assets/Rfig/PCA-2.svg" alt="plot of chunk PCA" /></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>        ggtitle<span class="p">(</span><span class="s">&#39;Mean expression vs dispersion&#39;</span><span class="p">)</span><span class="o">+</span>
        theme<span class="p">(</span> axis.title.x <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">14</span><span class="p">),</span> axis.text.x <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">12</span><span class="p">,</span>colour<span class="o">=</span><span class="s">&#39;black&#39;</span><span class="p">),</span>  
        axis.text.y <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">12</span><span class="p">,</span>colour<span class="o">=</span><span class="s">&#39;black&#39;</span><span class="p">),</span> axis.title.y <span class="o">=</span> element_text<span class="p">(</span>size<span class="o">=</span><span class="m">14</span><span class="p">))</span><span class="o">+</span>
        geom_smooth<span class="p">()</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text"><span></span>## Error in ggtitle(&quot;Mean expression vs dispersion&quot;) + theme(axis.title.x = element_text(size = 14), : non-numeric argument to binary operator</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>esMusSeur <span class="o">&lt;-</span> ScaleData<span class="p">(</span>object <span class="o">=</span> esMusSeur<span class="p">,</span>
        vars.to.regress <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;nUMI&quot;</span><span class="p">))</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text"><span></span>## [1] &quot;Regressing out nUMI&quot;
## 
  |                                                                       
  |                                                                 |   0%
  |                                                                       
  |=                                                                |   1%
  |                                                                       
  |=                                                                |   2%
  |                                                                       
  |==                                                               |   2%
  |                                                                       
  |==                                                               |   3%
  |                                                                       
  |==                                                               |   4%
  |                                                                       
  |===                                                              |   4%
  |                                                                       
  |===                                                              |   5%
  |                                                                       
  |====                                                             |   5%
  |                                                                       
  |====                                                             |   6%
  |                                                                       
  |====                                                             |   7%
  |                                                                       
  |=====                                                            |   7%
  |                                                                       
  |=====                                                            |   8%
  |                                                                       
  |======                                                           |   9%
  |                                                                       
  |======                                                           |  10%
  |                                                                       
  |=======                                                          |  10%
  |                                                                       
  |=======                                                          |  11%
  |                                                                       
  |========                                                         |  12%
  |                                                                       
  |========                                                         |  13%
  |                                                                       
  |=========                                                        |  13%
  |                                                                       
  |=========                                                        |  14%
  |                                                                       
  |=========                                                        |  15%
  |                                                                       
  |==========                                                       |  15%
  |                                                                       
  |==========                                                       |  16%
  |                                                                       
  |===========                                                      |  16%
  |                                                                       
  |===========                                                      |  17%
  |                                                                       
  |============                                                     |  18%
  |                                                                       
  |============                                                     |  19%
  |                                                                       
  |=============                                                    |  20%
  |                                                                       
  |=============                                                    |  21%
  |                                                                       
  |==============                                                   |  21%
  |                                                                       
  |==============                                                   |  22%
  |                                                                       
  |===============                                                  |  22%
  |                                                                       
  |===============                                                  |  23%
  |                                                                       
  |===============                                                  |  24%
  |                                                                       
  |================                                                 |  24%
  |                                                                       
  |================                                                 |  25%
  |                                                                       
  |=================                                                |  26%
  |                                                                       
  |=================                                                |  27%
  |                                                                       
  |==================                                               |  27%
  |                                                                       
  |==================                                               |  28%
  |                                                                       
  |===================                                              |  29%
  |                                                                       
  |===================                                              |  30%
  |                                                                       
  |====================                                             |  30%
  |                                                                       
  |====================                                             |  31%
  |                                                                       
  |====================                                             |  32%
  |                                                                       
  |=====================                                            |  32%
  |                                                                       
  |=====================                                            |  33%
  |                                                                       
  |======================                                           |  33%
  |                                                                       
  |======================                                           |  34%
  |                                                                       
  |=======================                                          |  35%
  |                                                                       
  |=======================                                          |  36%
  |                                                                       
  |========================                                         |  37%
  |                                                                       
  |=========================                                        |  38%
  |                                                                       
  |=========================                                        |  39%
  |                                                                       
  |==========================                                       |  39%
  |                                                                       
  |==========================                                       |  40%
  |                                                                       
  |==========================                                       |  41%
  |                                                                       
  |===========================                                      |  41%
  |                                                                       
  |===========================                                      |  42%
  |                                                                       
  |============================                                     |  42%
  |                                                                       
  |============================                                     |  43%
  |                                                                       
  |============================                                     |  44%
  |                                                                       
  |=============================                                    |  44%
  |                                                                       
  |=============================                                    |  45%
  |                                                                       
  |==============================                                   |  46%
  |                                                                       
  |==============================                                   |  47%
  |                                                                       
  |===============================                                  |  47%
  |                                                                       
  |===============================                                  |  48%
  |                                                                       
  |================================                                 |  49%
  |                                                                       
  |================================                                 |  50%
  |                                                                       
  |=================================                                |  50%
  |                                                                       
  |=================================                                |  51%
  |                                                                       
  |==================================                               |  52%
  |                                                                       
  |==================================                               |  53%
  |                                                                       
  |===================================                              |  53%
  |                                                                       
  |===================================                              |  54%
  |                                                                       
  |====================================                             |  55%
  |                                                                       
  |====================================                             |  56%
  |                                                                       
  |=====================================                            |  56%
  |                                                                       
  |=====================================                            |  57%
  |                                                                       
  |=====================================                            |  58%
  |                                                                       
  |======================================                           |  58%
  |                                                                       
  |======================================                           |  59%
  |                                                                       
  |=======================================                          |  59%
  |                                                                       
  |=======================================                          |  60%
  |                                                                       
  |=======================================                          |  61%
  |                                                                       
  |========================================                         |  61%
  |                                                                       
  |========================================                         |  62%
  |                                                                       
  |=========================================                        |  63%
  |                                                                       
  |==========================================                       |  64%
  |                                                                       
  |==========================================                       |  65%
  |                                                                       
  |===========================================                      |  66%
  |                                                                       
  |===========================================                      |  67%
  |                                                                       
  |============================================                     |  67%
  |                                                                       
  |============================================                     |  68%
  |                                                                       
  |=============================================                    |  68%
  |                                                                       
  |=============================================                    |  69%
  |                                                                       
  |=============================================                    |  70%
  |                                                                       
  |==============================================                   |  70%
  |                                                                       
  |==============================================                   |  71%
  |                                                                       
  |===============================================                  |  72%
  |                                                                       
  |===============================================                  |  73%
  |                                                                       
  |================================================                 |  73%
  |                                                                       
  |================================================                 |  74%
  |                                                                       
  |=================================================                |  75%
  |                                                                       
  |=================================================                |  76%
  |                                                                       
  |==================================================               |  76%
  |                                                                       
  |==================================================               |  77%
  |                                                                       
  |==================================================               |  78%
  |                                                                       
  |===================================================              |  78%
  |                                                                       
  |===================================================              |  79%
  |                                                                       
  |====================================================             |  79%
  |                                                                       
  |====================================================             |  80%
  |                                                                       
  |=====================================================            |  81%
  |                                                                       
  |=====================================================            |  82%
  |                                                                       
  |======================================================           |  83%
  |                                                                       
  |======================================================           |  84%
  |                                                                       
  |=======================================================          |  84%
  |                                                                       
  |=======================================================          |  85%
  |                                                                       
  |========================================================         |  85%
  |                                                                       
  |========================================================         |  86%
  |                                                                       
  |========================================================         |  87%
  |                                                                       
  |=========================================================        |  87%
  |                                                                       
  |=========================================================        |  88%
  |                                                                       
  |==========================================================       |  89%
  |                                                                       
  |==========================================================       |  90%
  |                                                                       
  |===========================================================      |  90%
  |                                                                       
  |===========================================================      |  91%
  |                                                                       
  |============================================================     |  92%
  |                                                                       
  |============================================================     |  93%
  |                                                                       
  |=============================================================    |  93%
  |                                                                       
  |=============================================================    |  94%
  |                                                                       
  |=============================================================    |  95%
  |                                                                       
  |==============================================================   |  95%
  |                                                                       
  |==============================================================   |  96%
  |                                                                       
  |===============================================================  |  96%
  |                                                                       
  |===============================================================  |  97%
  |                                                                       
  |===============================================================  |  98%
  |                                                                       
  |================================================================ |  98%
  |                                                                       
  |================================================================ |  99%
  |                                                                       
  |=================================================================| 100%
## Time Elapsed:  2.01855705181758 mins
## [1] &quot;Scaling data matrix&quot;
## 
  |                                                                       
  |                                                                 |   0%
  |                                                                       
  |=================================================================| 100%</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>esMusSeur <span class="o">&lt;-</span> RunPCA<span class="p">(</span>object <span class="o">=</span> esMusSeur<span class="p">,</span>
               pc.genes <span class="o">=</span> esMusSeur<span class="o">@</span>var.genes<span class="p">,</span>
               do.print <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span>
               pcs.print <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">5</span><span class="p">,</span>
               genes.print <span class="o">=</span> <span class="m">5</span><span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text"><span></span>## [1] &quot;PC1&quot;
## [1] &quot;Krt8&quot;   &quot;Krt18&quot;  &quot;S100a6&quot; &quot;Gsn&quot;    &quot;Lgals1&quot;
## [1] &quot;&quot;
## [1] &quot;Npm1&quot;   &quot;Dppa5a&quot; &quot;Ptma&quot;   &quot;Nlrp1a&quot; &quot;Rpl21&quot; 
## [1] &quot;&quot;
## [1] &quot;&quot;
## [1] &quot;PC2&quot;
## [1] &quot;Lrrc58&quot; &quot;Tpm1&quot;   &quot;Tagln&quot;  &quot;Eno1b&quot;  &quot;Eno1&quot;  
## [1] &quot;&quot;
## [1] &quot;Tdh&quot;     &quot;Mir690&quot;  &quot;Esrrb&quot;   &quot;Zfp42&quot;   &quot;Slc29a1&quot;
## [1] &quot;&quot;
## [1] &quot;&quot;
## [1] &quot;PC3&quot;
## [1] &quot;Gm15772&quot; &quot;Mt2&quot;     &quot;Esrrb&quot;   &quot;Tdh&quot;     &quot;Hmces&quot;  
## [1] &quot;&quot;
## [1] &quot;Acsf2&quot;   &quot;Ppm1k&quot;   &quot;Gm10941&quot; &quot;Agl&quot;     &quot;Mtfmt&quot;  
## [1] &quot;&quot;
## [1] &quot;&quot;
## [1] &quot;PC4&quot;
## [1] &quot;Klf5&quot;   &quot;Malat1&quot; &quot;Klf6&quot;   &quot;Slc2a3&quot; &quot;Sparc&quot; 
## [1] &quot;&quot;
## [1] &quot;Il12a&quot;    &quot;Gm20740&quot;  &quot;Olfr954&quot;  &quot;Fam229b&quot;  &quot;Tmem126b&quot;
## [1] &quot;&quot;
## [1] &quot;&quot;
## [1] &quot;PC5&quot;
## [1] &quot;Gapdh&quot;  &quot;Gpx4&quot;   &quot;Gm8709&quot; &quot;Dusp9&quot;  &quot;Gsn&quot;   
## [1] &quot;&quot;
## [1] &quot;Erdr1&quot;     &quot;Hist1h2ao&quot; &quot;Apoe&quot;      &quot;Hist1h2ap&quot; &quot;Flnb&quot;     
## [1] &quot;&quot;
## [1] &quot;&quot;</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>PrintPCAParams<span class="p">(</span>esMusSeur<span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-text" data-lang="text"><span></span>## Parameters used in latest PCA calculation run on: 2018-03-25 16:04:17
## =============================================================================
## PCs computed    Genes used in calculation    PCs Scaled by Variance Explained
##     20                  5707                               TRUE
## -----------------------------------------------------------------------------
## rev.pca 
##  FALSE
## -----------------------------------------------------------------------------
## Full gene list can be accessed using 
##  GetCalcParam(object = object, calculation = &quot;RunPCA&quot;, parameter = &quot;pc.genes&quot;)</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>VizPCA<span class="p">(</span>object <span class="o">=</span> esMusSeur<span class="p">,</span> pcs.use <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">)</span></code></pre></figure>

<p><img src="/assets/Rfig/PCA-3.svg" alt="plot of chunk PCA" /></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>PCAPlot<span class="p">(</span>object <span class="o">=</span> esMusSeur<span class="p">,</span> dim.1 <span class="o">=</span> <span class="m">1</span><span class="p">,</span> dim.2 <span class="o">=</span> <span class="m">2</span><span class="p">)</span></code></pre></figure>

<p><img src="/assets/Rfig/PCA-4.svg" alt="plot of chunk PCA" /></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>esMusSeur <span class="o">&lt;-</span> ProjectPCA<span class="p">(</span>object <span class="o">=</span> esMusSeur<span class="p">,</span> do.print <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span></code></pre></figure>

<p>By plotting the standard deviation for each principal compartment, I can see that the deviation stups at around the 12th compartment.</p>

<p>K-nearest neighbour finds the number of groups in the data set, which is used to colour the data in the tSNE-plot</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>PCElbowPlot<span class="p">(</span>esMusSeur<span class="p">)</span></code></pre></figure>

<p><img src="/assets/Rfig/subsetPCA-1.svg" alt="plot of chunk subsetPCA" /></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>esMusSeur <span class="o">&lt;-</span> FindClusters<span class="p">(</span>object <span class="o">=</span> esMusSeur<span class="p">,</span>
                     reduction.type <span class="o">=</span> <span class="s">&quot;pca&quot;</span><span class="p">,</span>
                     dims.use <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">,</span>
                     resolution <span class="o">=</span> <span class="m">0.6</span><span class="p">,</span>
                     print.output <span class="o">=</span> <span class="m">0</span><span class="p">,</span>
                     save.SNN <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
 
esMusSeur <span class="o">&lt;-</span> FindClusters<span class="p">(</span>object <span class="o">=</span> esMusSeur<span class="p">,</span>
                     reduction.type <span class="o">=</span> <span class="s">&quot;pca&quot;</span><span class="p">,</span>
                     dims.use <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">12</span><span class="p">,</span>
                     resolution <span class="o">=</span> <span class="m">0.6</span><span class="p">,</span>
                     print.output <span class="o">=</span> <span class="m">0</span><span class="p">,</span>
                     save.SNN <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>


esMusSeur <span class="o">&lt;-</span> RunTSNE<span class="p">(</span>object <span class="o">=</span> esMusSeur<span class="p">,</span>
                dims.use <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">12</span><span class="p">,</span>
                do.fast <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>

TSNEPlot<span class="p">(</span>object <span class="o">=</span> esMusSeur<span class="p">,</span> do.label <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span></code></pre></figure>

<p><img src="/assets/Rfig/subsetPCA-2.svg" alt="plot of chunk subsetPCA" /></p>

<p>I then want to identify the top genes per cluster.</p>

<p>The top genes can then be plotted as violin plots, feature plots, which projects the expression on the tSNE-plot, or as heatmaps for the top 5 highest expressed genes per cluster.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>VlnPlot<span class="p">(</span>object <span class="o">=</span> esMusSeur<span class="p">,</span> features.plot <span class="o">=</span> topMarkers<span class="o">$</span>gene<span class="p">,</span> point.size.use <span class="o">=</span> <span class="m">0</span><span class="p">)</span>

FeaturePlot<span class="p">(</span>object <span class="o">=</span> esMusSeur<span class="p">,</span>
            features.plot <span class="o">=</span> topMarkers<span class="o">$</span>gene<span class="p">,</span>
            cols.use <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="s">&quot;grey&quot;</span><span class="p">,</span> <span class="s">&quot;blue&quot;</span><span class="p">),</span>
            reduction.use <span class="o">=</span> <span class="s">&quot;tsne&quot;</span><span class="p">)</span></code></pre></figure>

<p><img src="/assets/Rfig/topGenes-1.svg" alt="plot of chunk topGenes" /><img src="/assets/Rfig/topGenes-2.svg" alt="plot of chunk topGenes" /></p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span></span>topMarkers5 <span class="o">&lt;-</span> esMusSeur.markers <span class="o">%&gt;%</span> 
    group_by<span class="p">(</span>cluster<span class="p">)</span> <span class="o">%&gt;%</span> 
    top_n<span class="p">(</span><span class="m">5</span><span class="p">,</span> avg_logFC<span class="p">)</span>
    
DoHeatmap<span class="p">(</span>object <span class="o">=</span> esMusSeur<span class="p">,</span>
          genes.use <span class="o">=</span> topMarkers5<span class="o">$</span>gene<span class="p">,</span>
          remove.key <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span>
          slim.col.label <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span></code></pre></figure>

<p><img src="/assets/Rfig/topGenes-3.svg" alt="plot of chunk topGenes" /></p>

<h3 id="there-you-go-my-first-single-cell-analysis-using-seurat-and-following-the-examples-from-dave-tang">There you go. My first single cell analysis using Seurat, and following the examples from Dave Tang</h3>


    </div>

    

  </div>

</article>

  </div>
</main>

<footer class="site-footer">
  <div class="container">
    <ul class="social">
  <li><a href="https://github.com/jespermaag" target="_blank"><i class="icon icon-github"></i></a></li>
  <li><a href="https://twitter.com/jespermaag" target="_blank"><i class="icon icon-twitter"></i></a></li>
  <li><a href="https://linkedin.com/in/jesper-maag" target="_blank"><i class="icon icon-linkedin"></i></a></li>
</ul>
    <p class="txt-medium-gray">
      <small>&copy;2020 All rights reserved. Made with <a href="http://jekyllrb.com/" target="_blank">Jekyll</a> and ♥</small>
    </p>
  </div>
</footer>


  <a href="http://jespermaag.github.io" target="_blank" class="github-corner"><svg width="80" height="80" viewBox="0 0 250 250" style="fill:#337ab7; color:#fff; position: absolute; top: 0; border: 0; right: 0;"><path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path><path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2" fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path><path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"></path></svg></a>
</body>
</html>
